From 1b220561f4d004410e1cbea41c14b8f42a618544 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Sun, 20 Jul 2014 11:53:10 +0100
Subject: [PATCH 1/6] SendReceiveStringList: handle the UNKNOWN_COMMAND
 response properly

Change it to return false and log the error rather than return true
and carry on as though nothing is wrong.

This should never happen in reality but if it does SendReceiveStringList
should not return true and carry on as normal since we don't handle that
well in a lot of places.
---
 mythtv/libs/libmythbase/mythcorecontext.cpp |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/mythtv/libs/libmythbase/mythcorecontext.cpp b/mythtv/libs/libmythbase/mythcorecontext.cpp
index b986ed7..0c6604f 100644
--- a/mythtv/libs/libmythbase/mythcorecontext.cpp
+++ b/mythtv/libs/libmythbase/mythcorecontext.cpp
@@ -1309,6 +1309,14 @@ bool MythCoreContext::SendReceiveStringList(
 
             ok = false;
         }
+        else if (strlist[0] == "UNKNOWN_COMMAND")
+        {
+            LOG(VB_GENERAL, LOG_ERR,
+                QString("Protocol query '%1' responded with the error 'UNKNOWN_COMMAND'")
+                        .arg(query_type));
+
+            ok = false;
+        }
     }
 
     return ok;
-- 
1.7.10.2

